pkg/telemetry/marketing/device.go

```diff
@@ -0,0 +1,63 @@
+package marketing
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/version"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const annotation = "stackrox.com/telemetry-apipaths"
+const orgID = "stackrox.com/organization-id"
+
+var config *Config
+
+// GetDeviceConfig collects the central instance telemetry configuration.
+func GetDeviceConfig() (*Config, error) {
+	if config != nil {
+		return config, nil
+	}
+	rc, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create k8s config")
+	}
+	clientset, err := kubernetes.NewForConfig(rc)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create k8s clientset")
+	}
+	v, err := clientset.ServerVersion()
+	if err != nil {
+		return nil, err
+	}
+	orchestrator := storage.ClusterType_KUBERNETES_CLUSTER.String()
+	if env.OpenshiftAPI.BooleanSetting() {
+		orchestrator = storage.ClusterType_OPENSHIFT_CLUSTER.String()
+	}
+
+	di := clientset.AppsV1().Deployments("stackrox")
+	opts := v1.GetOptions{}
+	d, err := di.Get(context.Background(), "central", opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot get central deployment")
+	}
+	paths := d.GetAnnotations()[annotation]
+
+	config = &Config{
+		ID:       string(d.GetUID()),
```

Comment:
What is this ID? Is it supposed to be the instance ID generated by fleet manager?

Reply:
This is central deployment UID, which essentially identifies a central deployment. Should work for both managed and wild installations.